owncloudApp/src/main/java/com/owncloud/android/presentation/ui/settings/SettingsActivity.kt

```diff
@@ -58,12 +63,40 @@ class SettingsActivity : AppCompatActivity() {
     override fun onOptionsItemSelected(item: MenuItem): Boolean {
         when (item.itemId) {
             android.R.id.home -> {
-                intent = Intent(this, FileDisplayActivity::class.java).apply {
-                    addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
+                if (supportFragmentManager.findFragmentByTag("logs") != null) {
+                    supportFragmentManager.popBackStack()
+                    supportActionBar?.setTitle(R.string.actionbar_settings)
```

Comment:
We can check directly if we are on root asking `backStackEntryCount` instead of checking tags 🍻 
```suggestion
                if (supportFragmentManager.backStackEntryCount > 0) {
                    supportFragmentManager.popBackStack()
                } else {
```